### PR TITLE
Add AllocationTick event test scenario

### PR DESCRIPTION
**Pull request by LakshanF**
Add a test scenario to #103198

### CODE REVIEW

**Comment by LakshanF** [comment]
The test is passing on the outer loop run, 

- Windows x64
  - https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-103290-merge-f2707f77726d4bba9b/tracing/1/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation/output.txt?helixlogtype=result
- Linux-x64
  - https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-103290-merge-383d0aa54ea64c249e/tracing/1/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation/output.txt?helixlogtype=result
- Linux ARM
  - https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-103290-merge-9e45f70a96654c00a5/tracing/1/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation/output.txt?helixlogtype=result
- Windows x86
  - https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-103290-merge-593f51b97c104dd9b1/tracing/1/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation/output.txt?helixlogtype=result

File: src/tests/tracing/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation.cs [hunk 1]
```diff
@@ -64,6 +64,9 @@ public static int TestEntryPoint()
                 RuntimeEventValidation eventValidation = new RuntimeEventValidation();
                 eventValidation = null;
                 GC.Collect();
+
+                // Explicitly targeting AllocationTick event
+                GC.KeepAlive(new int[1000]);
```
Comment (davmason) [inline src/tests/tracing/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation.cs:69]: Does the GC.KeepAlive do anything here? Is it just to make sure the compiler doesn't optimize it away completely?

Comment (LakshanF) [reply src/tests/tracing/eventpipe/simpleruntimeeventvalidation/simpleruntimeeventvalidation.cs:69]: yes, just to make sure.

**Review by jkotas — APPROVED** [review]
Thank you!

**Review by davmason — APPROVED** [review]

**Pull request merged**